rcl/src/rcl/node.c

```diff
@@ -163,6 +165,7 @@ rcl_node_init(
   ret = rmw_validate_namespace(local_namespace_, &validation_result, NULL);
   if (ret != RMW_RET_OK) {
     RCL_SET_ERROR_MSG(rmw_get_error_string_safe(), *allocator);
+    allocator->deallocate((char *)local_namespace_, allocator->state);
```

Comment:
This cannot be done if line 147 is executed (if `local_namespace_` is a literal string). I think it needs to check `should_free_local_namespace_`.

Here and below.

Reply:
exactly, thanks !